### PR TITLE
Navigation and process management

### PR DESCRIPTION
**Pull request by chicoxyzzy**
based on https://github.com/socketsupply/socket/pull/29

### CODE REVIEW

File: runtime.js [hunk 1]
```diff
@@ -133,6 +133,22 @@ export async function hide (index = window.__args.index) {
...
+ */
+export async function navigate ({ window: _window, value: _value }) {
+  const index = _window ?? window.__args.index
+  const value = `file://${window.__args.cwd()}/${_value}`
+  return await ipc.send('navigate', { index, value })
+}
+
+export async function openProcess () {
+  return await ipc.send('process.open')
+}
```
Comment (chicoxyzzy) [inline runtime.js]: I think we need to move this to a separate module. This way tree shaking will work better. WDYT about `backendProcess.js` naming? @heapwolf @jwerle
Related issue https://github.com/socketsupply/socket/issues/23

Comment (chicoxyzzy) [reply runtime.js]: or maybe just `backend`, so methods will be `io.backend.open({ force: boolean })`, `io.backend.kill()`, `io.backend.isOpen()`

Comment (chicoxyzzy) [reply runtime.js]: moved to `backend.js`

File: runtime.js [hunk 2]
```diff
@@ -133,6 +133,18 @@ export async function hide (index = window.__args.index) {
...
 
+/**
+ * @param {object} opts - an options object
+ * @param {number} [opts.window = window.__args.index] - the index of the window
+ * @param {number} opts.value - the path to the HTML file to load into the window
+ * @return {Promise<ipc.Result>}
+ */
+export async function navigate ({ window: _window, value: _value }) {
+  const index = _window ?? window.__args.index
+  const value = `file://${window.__args.cwd()}/${_value}`
```
Comment (jwerle) [inline runtime.js:144]: this assumes `cwd()` is always pointing to the resources directory of an app - is that true?

Comment (chicoxyzzy) [reply runtime.js:144]: yes, it's resources directory path

**Review by jwerle — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
this also needs https://github.com/socketsupply/socket/pull/29 to be merged first

**Pull request merged**